src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/managehearings/HearingsAboutToSubmitHandlerTest.java

```diff
@@ -85,21 +91,37 @@ void givenValidCaseData_whenHandle_thenHearingAddedToManageHearingsList() {
 
         FinremCaseData caseData = FinremCaseData.builder()
             .manageHearingsWrapper(ManageHearingsWrapper.builder()
+                .manageHearingsActionSelection(ManageHearingsAction.ADD_HEARING)
                 .workingHearing(hearingToAdd)
                 .build())
             .build();
 
         FinremCallbackRequest request = FinremCallbackRequestFactory.from(Long.parseLong(caseReference),
             CaseType.CONTESTED, caseData);
 
+        when(manageHearingsDocumentService.generateHearingNotice(hearingToAdd, request.getCaseDetails(), AUTH_TOKEN))
+            .thenReturn(CaseDocument
+                .builder()
+                .documentUrl("documentUrl")
+                .documentBinaryUrl("documentBinaryUrl")
+                .documentFilename("HearingNotice.pdf")
+                .uploadTimestamp(LocalDateTime.now())
+                .build()
+        );
+
         var response = manageHearingsAboutToSubmitHandler.handle(request, AUTH_TOKEN);
         var responseManageHearingsWrapper = response.getData().getManageHearingsWrapper();
+        var hearingDocumentAdded = responseManageHearingsWrapper.getHearingDocumentsCollection().getFirst();
+        var hearingId = responseManageHearingsWrapper.getHearings().getFirst().getId();
 
         assertThat(responseManageHearingsWrapper.getHearings())
             .extracting(ManageHearingsCollectionItem::getValue)
             .contains(hearingToAdd);
+        assertThat(hearingDocumentAdded.getValue().getHearingId()).isEqualTo(hearingId);
+        assertThat(hearingDocumentAdded.getValue().getHearingDocument().getDocumentFilename())
+            .isEqualTo("HearingNotice.pdf");
+        assertThat(responseManageHearingsWrapper.getWorkingHearingId()).isEqualTo(hearingId);
         assertThat(responseManageHearingsWrapper.getWorkingHearing()).isNull();
-        assertThat(responseManageHearingsWrapper.getManageHearingsActionSelection()).isNull();
     }
 
     private Hearing createHearingToAdd() {
```

Comment:
So performAddHearing could be implicitly tested here.  It's quite tricky to determine that.

performAddHearing is going to increase in complexity when notification types included for new hearing types.

Suggest biting the bullet now, and refactoring performAddHearing into a service as a public method so it can be tested.  Or put a big todo comment on this to say that will happen, with an accompanying test (so we can pick up in next PR).